packages/ide/src/client.ts

```diff
@@ -34,6 +34,14 @@ export abstract class IdeClient {
 		this.loadTime = time(2500);
 
 		let appWindow: Window | undefined;
+
+		window.addEventListener("beforeunload", (e) => {
```

Comment:
This should only fire if there is unsaved changes that needs to be pushed to the origin.

Reply:
The goal is to prevent accidentally closing the tab for those who have `ctrl+w` ingrained in muscle memory. Unfortunately I think this is the only way we can do it. :disappointed: